src/Avalonia.Base/Styling/StyleInstance.cs

```diff
@@ -1,137 +1,103 @@
-                _animationTrigger = new Subject<bool>();
-
-                for (var i = 0; i < animationsCount; ++i)
-                {
-                    _animations[i] = animations[i].Apply(animatable, null, _animationTrigger);
-                }
-            }
+            Priority = activator is object ? BindingPriority.StyleTrigger : BindingPriority.Style;
+            Source = style;
         }
 
-        public bool HasActivator => _activator is not null;
-        public bool IsActive { get; private set; }
+        public bool HasActivator => _activator is object;
+
         public IStyle Source { get; }
-        public IStyleable Target { get; }
 
-        public void Start()
+        bool IStyleInstance.IsActive => _isActive;
+        
+        public void Add(ISetterInstance instance)
         {
-            var hasActivator = HasActivator;
-
-            if (_setters is not null)
-            {
-                foreach (var setter in _setters)
-                {
-                    setter.Start(hasActivator);
-                }
-            }
-
-            if (hasActivator)
+            if (instance is IValueEntry valueEntry)
             {
-                _activator!.Subscribe(this, 0);
-            }
-            else if (_animationTrigger is not null)
-            {
-                _animationTrigger.OnNext(true);
+                if (Contains(valueEntry.Property))
+                    throw new InvalidOperationException(
+                        $"Duplicate setter encountered for property '{valueEntry.Property}' in '{Source}'.");
+                Add(valueEntry);
             }
+            else
+                (_setters ??= new()).Add(instance);
         }
 
-        public void Dispose()
+        public void Add(IList<IAnimation> animations)
         {
-            if (_setters is not null)
-            {
-                foreach (var setter in _setters)
-                {
-                    setter.Dispose();
-                }
-            }
+            if (_animations is null)
+                _animations = new List<IAnimation>(animations);
+            else
+                _animations.AddRange(animations);
+        }
 
-            if (_animations is not null)
+        public void ApplyAnimations(AvaloniaObject control)
+        {
+            if (_animations is not null && control is Animatable animatable)
             {
-                foreach (var subscription in _animations)
-                {
-                    subscription.Dispose();
-                }
+                _animationTrigger ??= new Subject<bool>();
+                foreach (var animation in _animations)
+                    animation.Apply(animatable, null, _animationTrigger);
             }
+        }
 
+        public override void Dispose()
+        {
+            base.Dispose();
             _activator?.Dispose();
         }
 
-        private void ActivatorChanged(bool value)
+        public new void MakeShared() => base.MakeShared();
```

Comment:
Is it a debug leftover?

Reply:
Nope, base implementation is `protected`